mcp/session.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2025 The Go MCP SDK Authors. All rights reserved.
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+package mcp
+
+import (
+	"context"
+	"io/fs"
+	"sync"
+)
+
+// SessionState is the state of a session.
+type SessionState struct {
+	// InitializeParams are the parameters from the initialize request.
+	InitializeParams *InitializeParams `json:"initializeParams"`
+
+	// LogLevel is the logging level for the session.
+	LogLevel LoggingLevel `json:"logLevel"`
+
+	// TODO: resource subscriptions
+}
```

Comment:
Do we want to store the Initialized state of the session along side the `InitializeParams`, or is there another plan for tackling that?

While the `InitializeParams` and `LogLevel` can be loaded from an external store, the `ServerSession.initialized` bool is still private. If the request comes into a server that did not initialize the session, any subsequent requests can't be fulfilled until the session goes through the Initialization process again.

https://github.com/omgitsads/go-sdk/commit/9f888732d5b98cf7291ddf4b6bc72e3db33929c2 moves the intialized bool into the `SessionState`, but not sure how you'd want to tackle that happening before the response has been sent?

Reply:
I can confirm having the same issue after trying quick implementation on top of the PR new interfaces. 

If we emulate the scenario when a client initialised a connection and retrieved session id, if on a subsequent request, let's say hitting a particular tool,  when it hits a new instance (assuming it lost the connection, or it works in simple request/response mode behind load balancers + mesh) new instance has no awereness on initialised state = it is equal false, so regardless of successful session retrieval a new instance will reject the request.

https://github.com/modelcontextprotocol/go-sdk/blob/0375535db46aa5b2117f4065a20a5d594ce8860d/mcp/server.go#L755

Reply:
Added the initialized bool to SessionState.